src/dune_config_file/dune_config_file.ml

```diff
@@ -53,6 +53,14 @@ module Dune_config = struct
     ;;
   end
 
+  module Pkg_enabled = struct
+    type t = bool
+
+    let decode = enum' [ "enabled", return true; "disabled", return false ]
```

Comment:
Anticipating that we may want to add more data to a `pkg` stanza in the future, should this maybe be in a field like `(pkg (enabled true))`?

As we discussed, we could imagine eventually wanting to move the other stanzas that only make sense in the context of dune package managemen't under a `pkg` stanza, or putting them in a separate `dune-pkg` file. Introducing this stanza in a way that will make that easier could be wise then perhaps?

Reply:
I can appreciate your perspective. IMO, it is worse complexity to have several different ways of writing a configuration than to have one uniform way that is slightly more verbose, and it is better to introduces changes in a way that allows for easy backward compatible changes for stuff we are very likely to do, than to have to rely on a lot of deprecation, versioning, and complications.

But I take your point and I'm happy to disagree and commit to the way you think is best here.

Reply:
> it is better to introduces changes in a way that allows for easy backward compatible changes for stuff we are very likely to do

But are we? We've gone some two years without this, `pins`, `depopts` etc are configured elsewhere anyway, so I am not sure there's any extensions coming up soon.

Also note that `enabled` and `disabled` are not `true`/`false`. The dune codebase has a lot of cases where the option set is not binary. e.g. `only-dev-tools` could be a possible option as well without having to make the default case a record. I am trying to adhere to YAGNI but of course it's debatable where to cut the line.

But I'd say it is somewhat bikeshedding. I can implement the other way as well if people feel strongly.

@gridbugs What's your take on this?